qcodes/tests/test_station.py

```diff
@@ -201,3 +215,306 @@ def test_station_after_instrument_is_closed():
     with pytest.raises(KeyError, match='Component bob is not part of the '
                                        'station'):
         station.remove_component('bob')
+
+@pytest.fixture
+def example_station_config():
+    """
+    Returns path to temp yaml file with station config.
+    """
+    sims_path = get_qcodes_path('instrument', 'sims')
+    test_config = f"""
+instruments:
+  lakeshore:
+    driver: qcodes.instrument_drivers.Lakeshore.Model_336
+    type: Model_336
+    enable_forced_reconnect: true
+    address: GPIB::2::65535::INSTR
+    init:
+      visalib: '{sims_path}lakeshore_model336.yaml@sim'
+  mock_dac:
+    driver: qcodes.tests.instrument_mocks
+    type: DummyInstrument
+    enable_forced_reconnect: true
+    init:
+      gates: {{"ch1", "ch2"}}
+  mock_dac2:
+    driver: qcodes.tests.instrument_mocks
+    type: DummyInstrument
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        filename = Path(tmpdirname, 'station_config.yaml')
+        with filename.open('w') as f:
+            f.write(test_config)
+        yield str(filename)
+
+def station_from_config_str(config: str) -> Station:
+    st = Station(config_file=None)
+    st.load_config(config)
+    return st
+
+
+def has_station_config_been_loaded(st: Station) -> bool:
+    return "StationConfigurator" in st.components.keys()
+
+
+@pytest.fixture
+def example_station(example_station_config):
+    return Station(config_file=example_station_config)
+
+
+# instrument loading related tests
+def test_station_config_path_resolution(example_station_config):
```

Comment:
You are absolutely right. I was feeling pressed in time so I did not add those comments. This is an excellent example of going slow to move fast: without me adding the comments it will take all others a lot longer to review... Thanks again Mikail!